File: src/edu/gbcg/DBSelector/RedditSubmission/SubmissionSetMapper.java
package edu.gbcg.DBSelector.RedditSubmission;

import edu.gbcg.DBSelector.RSMapper;
import edu.gbcg.dbcreator.Reddit.Submissions;

import java.sql.ResultSet;
import java.util.List;
import java.util.Map;

public class SubmissionSetMapper extends RSMapper {
    public SubmissionSetMapper(Map<String, String> map){ super(map); }
    public SubmissionSetMapper(){ super(); }
    public List<RSMapper> buildMappers(ResultSet rs){
        return buildMappers_impl(rs, Submissions.getColumnsForDB());
    }
}

File: src/main/kotlin/edu/antevortadb/dbInteraction/dbSelector/reddit/submissions/SubmissionSetMapper.kt
/*
 * Copyright (c) 2018 Sean Grimes. All Rights Reserved.
 * License: MIT
 */

package edu.antevortadb.dbInteraction.dbSelector.reddit.submissions

import edu.antevortadb.dbInteraction.columnsAndKeys.RedditSubs
import edu.antevortadb.dbInteraction.dbSelector.RSMapper
import org.json.JSONObject
import java.sql.ResultSet

@Suppress("unused")
class SubmissionSetMapper: RSMapper {
    constructor(map: Map<String, String>): super(map)
    constructor(jsonObject: JSONObject): super(jsonObject)
    constructor(): super()
    override fun buildMappers(rs: ResultSet): MutableList<RSMapper> {
        return buildMappersImpl(rs, RedditSubs.columnNames())
    }
}
